File: src/main/java/io/github/emcw/caching/CacheStrategy.java
package io.github.emcw.caching;

/**
 * Specifies the caching strategy to use.<b></b>
 * <p><b>Types: </b> {@link #LAZY}, {@link #TIME_BASED}
 */
public enum CacheStrategy {
    /**
     * Only update on next call if expired (expiry should be set).
     */
    LAZY,
    /**
     * Always update at an interval (expiry not needed).
     */
    TIME_BASED
}

File: src/main/kotlin/io/github/emcw/caching/CacheStrategy.kt
package io.github.emcw.caching

/**
 * Specifies the caching strategy to use.****
 *
 * **Items: ** [.LAZY], [.TIME_BASED], [.HYBRID]
 */
enum class CacheStrategy {
    /**
     * Only update on next call (expiry should be set)
     */
    LAZY,

    /**
     * Always update at an interval (expiry not needed)
     */
    TIME_BASED,

    /**
     * Time based, but forces update if called before interval.
     */
    HYBRID
}